docs(how-to): rpn-san (#219)
scaleway/docs-content: dedibox-network/rpn/how-to/mount-rpn-san-linux.mdx

description: This page explains how to mount a RPN-SAN volume on Linux
dates:
  validation: 2021-11-10
  posted: 2021-11-10
---

You can mount RPN-SAN volumes on Linux machines using Open-iSCSI. 

<Message type="requirement">

* You have an account and are logged into the [Dedibox Console](https://console.online.net)
* You have a Dedibox dedicated server
* You have a RPN-SAN 

</Message>

## How to install and configure Open-iSCSI

1. Update the apt package cache and install Open-iSCSI:
    ```
    apt update && apt install -y open-iscsi
    ```
2. Open the file `/etc/iscsi/iscsid.conf` in a text editor:
    ```
    nano /etc/iscsi/iscsid.conf
    ```

3. Add the following configuration lines to the configuration: 
    ```
    node.conn[0].timeo.noop_out_interval = 0
    node.conn[0].timeo.noop_out_timeout = 0
    node.session.timeo.replacement_timeout = 86400
    ```

    The purpose of this modification is to prevent your file system from becoming read-only if the link is interrupted for more than two minutes (by default) between your server and the storage server.

4. Adjust the setting for automatic reconnection of targets
    ```
    node.startup = automatic
    ````

> 5. Change the following settings to benefit from the data integrity check, if available

Reviewer: ```suggestion
5. Change the following settings to benefit from the data integrity check, if available:
```

    <Message type="note">

    The `SAN_SERVER` address is displayed in your Dedibox console.

    </Message>

    The target name should look like the following example: 
    ```
    X.X.X.X:3260,1 iqn.2013-01.net.online:XXXXXXXXXXX
    ```

2. Connect to the target by typing the following command: 
    ```
    iscsiadm -m node -T iqn.2013-01.net.online:XXXXXXXXXXX --login
    ```

3. Type `dmesg` to see the kernel logs. You should see some lines like in the following example
    ```
    scsi25 : iSCSI Initiator over TCP/IP
    scsi 25:0:0:0: Direct-Access     IET      VIRTUAL-DISK     0    PQ: 0 ANSI: 4
    sd 25:0:0:0: Attached scsi generic sg4 type 0
    sd 25:0:0:0: [sdd] 2147483648 512-byte logical blocks: (1.09 TB/1.00 TiB)
    sd 25:0:0:0: [sdd] Write Protect is off
    sd 25:0:0:0: [sdd] Mode Sense: 77 00 00 08
    sd 25:0:0:0: [sdd] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
    sdd: sdd1
    sd 25:0:0:0: [sdd] Attached SCSI disk
    ```
    In the example above, the disk detected is `sdd`, available at `/dev/sdd`. 

    <Message type="note"> 

    The assigned device name may be different, depending on your configuration.

    </Message>

>     You can now use your SAN disk as any other disk.

Reviewer: ```suggestion
    You can now use your SAN disk like you would any other disk.
```

## Important recommendations for use

As seen in the exaple abouve, the disk is available as `/dev/sdd.` It is not guaranteed that at the next reboot the same drive letter will be used, especially if you add other targets, or if your hardware configuration changes. It is therefore recommended not to use the names `/dev/sdX` in your configuration files.

### How to address devices when using LVM

> LVM automatically uses UUIDs to recognize the partitions it uses. No problem in this case.

Reviewer: ```suggestion
LVM automatically uses UUIDs to recognize the partitions it uses. In this case, no problems should arise.
```

### How to address devices when using direct disk formatting

> <Message typem="note">

Reviewer: ```suggestion
<Message type="note">
```

> This method is not recommended, because it does not allow you to evolve afterwards, by aggregating several RPN-SAN storage spaces.

Reviewer: ```suggestion
This method is not recommended, because it does not allow you to make changes afterwards, by aggregating several RPN-SAN storage spaces.
```

</Message>

> If you format directly the device or a partition of it, use the link created by the system in `/dev/disk/by-uuid/` or the fstab notation `UUID=XXX-XXX...` to address it.

Reviewer: ```suggestion
If you format the device or one of its partitions directly, use the link created by the system in `/dev/disk/by-uuid/` or the fstab notation `UUID=XXX-XXX...` for addressing purposes.
```

### How to address devices direct use of the device

> If your application does not support (unlike LVM) the auto-detection of the partitions associated with it via the UUID, it is recommended to use the link in `/dev/disk/by-id/`.

Reviewer: ```suggestion
If (unlike LVM) your application does not support the auto-detection of the partitions associated with it via the UUID, it is recommended to use the link in `/dev/disk/by-id/`.
```

> This is the available path that will be the most stable over time.

Reviewer: ```suggestion
This is the most stable available path.
```

---
meta:
  title: How to mount a RPN SAN volume on Linux
  description: This page explains how to mount a RPN SAN volume on Linux
content:
  h1: How to mount a RPN SAN volume on Linux
  paragraph: This page explains how to mount a RPN SAN volume on Linux
dates:
  validation: 2021-11-10
  posted: 2021-11-10
---

You can mount RPN SAN volumes on Linux machines using Open-iSCSI. 

<Message type="requirement">

* You have an account and are logged into the [Dedibox Console](https://console.online.net)
* You have a Dedibox dedicated server
* You have a RPN SAN 

</Message>

## How to install and configure Open-iSCSI

1. Update the apt package cache and install Open-iSCSI:
    ```
    apt update && apt install -y open-iscsi
    ```
2. Open the file `/etc/iscsi/iscsid.conf` in a text editor:
    ```
    nano /etc/iscsi/iscsid.conf
    ```

> 3. Add the following configuration lines to the configuration: 

Reviewer 2: ```suggestion
3. Add the following configuration lines: 
```

    The target name should look like the following example: 
    ```
    X.X.X.X:3260,1 iqn.2013-01.net.online:XXXXXXXXXXX
    ```

2. Connect to the target by typing the following command: 
    ```
    iscsiadm -m node -T iqn.2013-01.net.online:XXXXXXXXXXX --login
    ```

3. Type `dmesg` to see the kernel logs. You should see some lines like in the following example
    ```
    scsi25 : iSCSI Initiator over TCP/IP
    scsi 25:0:0:0: Direct-Access     IET      VIRTUAL-DISK     0    PQ: 0 ANSI: 4
    sd 25:0:0:0: Attached scsi generic sg4 type 0
    sd 25:0:0:0: [sdd] 2147483648 512-byte logical blocks: (1.09 TB/1.00 TiB)
    sd 25:0:0:0: [sdd] Write Protect is off
    sd 25:0:0:0: [sdd] Mode Sense: 77 00 00 08
    sd 25:0:0:0: [sdd] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
    sdd: sdd1
    sd 25:0:0:0: [sdd] Attached SCSI disk
    ```
    In the example above, the disk detected is `sdd`, available at `/dev/sdd`. 

    <Message type="note"> 

    The assigned device name may be different, depending on your configuration.

    </Message>

    You can now use your SAN disk like you would any other disk.

## Important recommendations for use

> As seen in the exaple abouve, the disk is available as `/dev/sdd.` It is not guaranteed that at the next reboot the same drive letter will be used, especially if you add other targets, or if your hardware configuration changes. It is therefore recommended not to use the names `/dev/sdX` in your configuration files.

Reviewer 2: ```suggestion
As seen in the example above, the disk is available as `/dev/sdd.` It is not guaranteed that at the next reboot the same drive letter will be used, especially if you add other targets, or if your hardware configuration changes. It is therefore recommended not to use the names `/dev/sdX` in your configuration files.
```

### How to address devices when using LVM
LVM automatically uses UUIDs to recognize the partitions it uses. In this case, no problems should arise.

### How to address devices when using direct disk formatting

<Message type="note">

> This method is not recommended, because it does not allow you to make changes afterwards, by aggregating several RPN SAN storage spaces.

Reviewer 2: ```suggestion
This method is not recommended, because aggregates several RPN SAN storage spaces, and does not allow you to make changes afterwards.
```

</Message>

If you format the device or one of its partitions directly, use the link created by the system in `/dev/disk/by-uuid/` or the fstab notation `UUID=XXX-XXX...` for addressing purposes.

> ### How to address devices direct use of the device

Reviewer 2: I dont understand this title

    The target name should look like the following example: 
    ```
    X.X.X.X:3260,1 iqn.2013-01.net.online:XXXXXXXXXXX
    ```

2. Connect to the target by typing the following command: 
    ```
    iscsiadm -m node -T iqn.2013-01.net.online:XXXXXXXXXXX --login
    ```

3. Type `dmesg` to see the kernel logs. You should see some lines like in the following example
    ```
    scsi25 : iSCSI Initiator over TCP/IP
    scsi 25:0:0:0: Direct-Access     IET      VIRTUAL-DISK     0    PQ: 0 ANSI: 4
    sd 25:0:0:0: Attached scsi generic sg4 type 0
    sd 25:0:0:0: [sdd] 2147483648 512-byte logical blocks: (1.09 TB/1.00 TiB)
    sd 25:0:0:0: [sdd] Write Protect is off
    sd 25:0:0:0: [sdd] Mode Sense: 77 00 00 08
    sd 25:0:0:0: [sdd] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
    sdd: sdd1
    sd 25:0:0:0: [sdd] Attached SCSI disk
    ```
    In the example above, the disk detected is `sdd`, available at `/dev/sdd`. 

    <Message type="note"> 

    The assigned device name may be different, depending on your configuration.

    </Message>

    You can now use your SAN disk like you would any other disk.

## Important recommendations for use

> As seen in the exaple abouve, the disk is available as `/dev/sdd.` It is not guaranteed that at the next reboot the same drive letter will be used, especially if you add other targets, or if your hardware configuration changes. It is therefore recommended not to use the names `/dev/sdX` in your configuration files.

Reviewer: ```suggestion
As seen in the exaple above, the disk is available as `/dev/sdd.` It is not guaranteed that at the next reboot the same drive letter will be used, especially if you add other targets, or if your hardware configuration changes. It is therefore recommended not to use the names `/dev/sdX` in your configuration files.
```
